hostinit/runner.go

```diff
@@ -53,32 +54,43 @@ func (r *Runner) Run(ctx context.Context, config *evergreen.Settings) error {
 		"GUID":    init.GUID,
 	})
 
-	if err := init.startHosts(ctx); err != nil {
```

Comment:
use a grip.Catcher to aggregate errors across threads `grip.NewSimpleCatcher()` and then return before line 94 if there was an error so that we don't bump the "last successful run time" in the case of an error. 

lgtm otherwise